drivers/i2s/i2s_ll_stm32.c

```diff
@@ -837,37 +857,46 @@ struct queue_item rx_1_ring_buf[CONFIG_I2S_STM32_RX_BLOCK_COUNT + 1];
 struct queue_item tx_1_ring_buf[CONFIG_I2S_STM32_TX_BLOCK_COUNT + 1];
 
 static struct i2s_stm32_data i2s_stm32_data_1 = {
-	.dma_name = I2S1_DMA_NAME,
 	.rx = {
-		.dma_channel = I2S1_DMA_CHAN_RX,
+		.dma_name = DT_I2S_1_DMA_CONTROLLER_RX,
+		.dma_channel = DT_I2S_1_DMA_CHANNEL_RX,
 		.dma_cfg = {
 			.block_count = 1,
-			.dma_slot = I2S1_DMA_SLOT_RX,
+			.dma_slot = DT_I2S_1_DMA_SLOT_RX,
 			.channel_direction = PERIPHERAL_TO_MEMORY,
 			.source_data_size = 1,  /* 16bit default */
 			.dest_data_size = 1,    /* 16bit default */
```

Comment:
Shouldn't source and dest data size be put equal to 2 as we said? This is number of bytes, so 16bit == 2 bytes.
An I think this commit should be squashed together with the commit for new DMA (e186b3ac), as they are inter-dependent (new DMA requires these changes, old DMA does not require these changes).

Do you agree?

Reply:
I was making the changes. But before this, I want to write a macro like the one used in this driver https://github.com/cybertale/zephyr/blob/62dce55d73a2c2ca9ccd93e10c8d5ba927c38050/drivers/dma/dma_stm32.c#L533 to save some time since declaration code for I2S1-I2S5 has too many copies.

Reply:
I was planning to do this in a new PR, but as you said only the new driver need this change, I will append them to this PR.

Reply:
Sure, it looks a very good idea to have such a macro!!